packages/user-group-spa/src/components/Group.test.js

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { render, waitForElementToBeRemoved, screen } from "@testing-library/react";
+import { MemoryRouter } from 'react-router-dom';
+import Group from "./Group";
+import BreadcrumbContextProvider from "../context/BreadcrumbContext";
+import Router from "react-router-dom";
+import { server } from "../mocks/server.js";
+import { rest } from "msw";
+import { groupData } from '../mocks/data';
+
+jest.mock( "react-router-dom", () => ( {
+  ...jest.requireActual("react-router-dom"),
+  useParams: jest.fn(),
+} ) );
+
+test( 'renders group information', async () => {
+  server.use(
+    rest.post("http://localhost:8080/graphql", (req, res, ctx) => {
+      return res(
+        ctx.status(200),
+        ctx.json(groupData)
+      );
+    })
+  );
+  // Mocking and returning mocked value from useParam() of react-router-dom
+  jest.spyOn(Router, "useParams").mockReturnValue({ cn: "one-platform" });
+
+  const { getByText, container } = render(
+    <MemoryRouter>
```

Comment:
Is there any reason we are wrapping every component with the provider and memory router

Reply:
@akhilmhdh Every component updates breadcrumb(in `useEffect()`) so the provider wrapping is needed. And to provide the router context it's recommended to wrap unit tests in one of the Router components hence wrapping`MemoryRouter`.